src/Symfony/Component/VarDumper/Server/DumpServer.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Server;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\VarDumper\Cloner\Data;
+
+/**
+ * A server collecting Data clones sent by a ServerDumper.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @final
+ */
+class DumpServer
+{
+    const HOST_ENV_VAR = 'VAR_DUMPER_SERVER';
+    const DEFAULT_HOST = 'tcp://0.0.0.0:9912';
+
+    private $host;
+    private $socket;
+    private $logger;
+
+    /**
+     * @param string|null          $host   The server host or null to read it from the VAR_DUMPER_SERVER env var
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct($host = null, LoggerInterface $logger = null)
+    {
+        if (null === $host) {
+            $host = getenv(static::HOST_ENV_VAR) ?: static::DEFAULT_HOST;
+        }
+
+        if (false === strpos($host, '://')) {
+            $host = 'tcp://'.$host;
+        }
+
+        $this->host = $host;
+        $this->logger = $logger;
+    }
+
+    public function start()
+    {
+        if (!$this->socket = stream_socket_server($this->host, $errno, $errstr)) {
+            throw new \RuntimeException(sprintf('Server start failed on "%s": %s %s.', $this->host, $errstr, $errno));
+        }
+    }
+
+    public function listen(callable $callback)
+    {
+        if (null === $this->socket) {
+            $this->start();
+        }
+
+        foreach ($this->getMessages() as $clientId => $message) {
+            $payload = unserialize(base64_decode($message));
+
+            // Impossible to decode the message, give up.
+            if (false === $payload) {
```

Comment:
`base64_decode` won't fail on a non base64 encoded string AFAIK, but `serialize` will trigger a notice. I think silencing it is enough but let me know.